src/app/reporting/SynchronizedReportSchedulerImpl.cpp

```diff
@@ -206,12 +201,14 @@ void SynchronizedReportSchedulerImpl::TimerFired()
 
     if (firedEarly)
     {
+        // If we fired the timer early, we need to recalculate the next report timeout and reschedule the report
         Timeout timeout = Milliseconds32(0);
         ReturnOnFailure(CalculateNextReportTimeout(timeout, nullptr, now));
         ScheduleReport(timeout, nullptr, now);
     }
     else
     {
+        // If we did not fire the timer early, we can schedule an engine run
```

Comment:
The key part here is not the "did not fire early", really.  It's "we have a handler that's reportable now", no?

Reply:
Yes it is.